server/controllers/accommodationController.js

```diff
@@ -0,0 +1,107 @@
+import AccommodationHelper from '../helpers/accommodationHelper';
+
+/**
+ * This class contains all methods
+ * required to handle all
+ * accommodation-related operations.
+ */
+class Accommodation {
+  /**
+   * This method handles the register request.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the user.
+   */
+  static async registerAccomodation(req, res) {
+    try {
+      const {
+        name,
+        description,
+        locationId,
+        amenities,
+        images,
+        rooms
+      } = req.body;
+
+      const { id, role } = req.user;
+      name.toUpperCase();
+
+      const createdBy = id;
+      if (role === 'Supplier' || role === 'Travel administrator') {
+        const accommodationFound = await AccommodationHelper.findAccommodation({ name });
+        if (accommodationFound) {
+          return res.status(409).json({
+            status: 409,
+            message: 'Accommodation facility already exists'
```

Comment:
Dynamically, specify the accommodation name in the response.

Reply:
Feedback implemented